kyo-direct/shared/src/main/scala/kyo/internal/AsyncShift.scala

```diff
@@ -16,10 +18,32 @@ end asyncShiftLowPriorityImplicit1
 
 object asyncShift extends asyncShiftLowPriorityImplicit1:
 
-    transparent inline given shiftedChunk[A]: ChunkAsyncShift[A] = new ChunkAsyncShift[A]
+    transparent inline given shiftedChunk[A]: ChunkAsyncShift[A]         = new ChunkAsyncShift[A]
+    transparent inline given shitferStream[V, S]: StreamAsyncShift[V, S] = new StreamAsyncShift
 
 end asyncShift
 
+class StreamAsyncShift[V, S] extends AsyncShift[Stream[V, S]]:
+    def map[F[_], V2](stream: Stream[V, S], monad: CpsMonad[F])(f: V => F[V2])(using
+        t1: Tag[Emit[Chunk[V]]],
+        t2: Tag[Emit[Chunk[V2]]],
+        ev: NotGiven[V2 <:< (Any < Nothing)],
+        fr: Frame
+    ): F[Stream[V2, S]] =
+        monad match
+            case _: KyoCpsMonad[?] => monad.pure(stream.map(a => f(a)).asInstanceOf[Stream[V2, S]])
+    end map
+
+    def filter[F[_]](stream: Stream[V, S], monad: CpsMonad[F])(f: V => F[Boolean])(using
+        tag: Tag[Emit[Chunk[V]]],
+        discr: Discriminator,
+        frame: Frame
+    ): F[Stream[V, S]] =
+        monad match
+            case _: KyoCpsMonad[?] => monad.pure(stream.filter(f).asInstanceOf[Stream[V, S]])
```

Comment:
It's incorrect. It works for simple use cases, when you unwrap after, for other cases, that's incorrect.

Funny thing, it would run in the same issues as in https://github.com/rcardin/yaes/discussions/33.

I will write a test to show how unsafe it is, then fix to something better.